lib/main.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2012-2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* jshint node: true */
+'use strict';
+
+var _ = require('lodash');
+var bedrock = require('bedrock');
+var BedrockError = bedrock.util.BedrockError;
+var config = bedrock.config;
+var brPermission = require('bedrock-permission');
+var brPassport = require('bedrock-passport');
+var database = require('bedrock-mongodb');
+var ensureAuthenticated = brPassport.ensureAuthenticated;
+var validate = require('bedrock-validation').validate;
+
+require('bedrock-express');
+
+// load config
+require('./config');
+
+bedrock.events.on('bedrock.configure', function() {
+  config.permission.roleBaseUrl =
+    config.server.baseUri + config['permission-http'].routes.roles;
+});
+
+bedrock.events.on('bedrock-express.configure.routes', function(app) {
+  var permissionsBasePath = config['permission-http'].routes.permissions;
+  var permissionsBaseUrl = config.server.baseUri + permissionsBasePath;
+  var rolesBasePath = config['permission-http'].routes.roles;
+
+  /**
+   * Retrieves all permissions from the config held permissions.
+   */
+  app.get(permissionsBasePath, ensureAuthenticated, function(req, res, next) {
+    var permissions = config.permission.permissions;
+    res.json(permissions);
+  });
+
+  /**
+   * Gets single role
+   */
+  app.get(
+    rolesBasePath + '/:id', ensureAuthenticated, function(req, res, next) {
+    var roleId = req.params.id;
+    if(config.permission.roleBaseUrl.length !== 0) {
+      roleId = config.permission.roleBaseUrl + '/' + roleId;
```

Comment:
This conditional seems odd here. Should it just always be prepending `rolesBasePath`? cc: @davidlehn thoughts?